### PR TITLE
(cabal init) Suggest BSD-3-Clause as default

### LINKED ISSUE #8757: `cabal init` should suggest a default license
**Describe the bug**

`cabal init` does not have a default license in its interactive menu. 

**To Reproduce**

Type `cabal init`, select options until you arrive at the license part, press Enter.

```
Please choose a license:
   1) BSD-2-Clause
   2) BSD-3-Clause
   3) Apache-2.0
   4) MIT
  ⁝
  14) LGPL-2.1-or-later
  15) LGPL-3.0-or-later
  16) AGPL-3.0-or-later
  17) Other (specify)
Your choice? ⏎
Empty input is not a valid choice.
```

**Expected behavior**
`cabal` to pick a license for me. I am partial to copyleft ones, but even MIT (permissive) would be better than “no default”.

`cabal` 3.9

### PR DESCRIPTION
**Pull request by ffaf1**
Closes #8757 

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: changelog.d/issue-8757 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+synopsis: Suggest BSD-3 as default license
```
Comment (andreasabel) [inline changelog.d/issue-8757:1]: Should this say something like?
```
synopsis: cabal init: suggest BSD-3 as default license
```

**Review by andreasabel — APPROVED** [review]

**Review by ulysses4ever — APPROVED** [review]
Thanks!

I believe Andreas' suggestion about the changelog should be implemented, though.

File: changelog.d/issue-8757 (hunk 1 above)
Comment (ffaf1) [reply changelog.d/issue-8757]: good suggersion, done

**Review by Kleidukos — APPROVED** [review]

**Comment by jneira** [comment]
another feature without regression tests? 😔

**Comment by ffaf1** [comment]
@jneira I can write one for `cabal-tests`, but it is going to be quite wobbly (breaking at minor `cabal init` changed), wouldn’ it?

**Comment by Mikolaj** [comment]
@jneira: please keep reminding us; this is so important

**Comment by jneira** [comment]
hmm cant check the code right now but ideally we should assert what we are suggesting to user interactively
but well, this is a tiny change, bigger ones were merged without tests 😝

**Review by jneira — APPROVED** [review]

**Comment by Mikolaj** [comment]
noooooo! don't give up!

**Comment by ffaf1** [comment]
That is a good suggestion. I will pause mergify and see if I can write an appropriate test. Ideally it should only check for a

     *  2) BSD-3-Clause

line after `cabal init` and 5 ⏎.

**Comment by Mikolaj** [comment]
I meant @jneira should not sell out for convenience and kindness of principles and correctness. :D 

Regarding tests, or perhaps @jneira meant that we could test that the init subsystem, with the newly instituted default chosen, produces sensible results (e.g., bsd3 licence in a record of init outcome)? E.g., run `cabal init --license bds3 --all-yes` and check if the result has bsd3 set. Somehow.

Of course, then the test is 1000 times longer than the PR, but that was true from the start. :D

**Comment by Mikolaj** [comment]
I see. So just run `cabal init --non-interactive` and golden-test all files generated. The simplest imaginable way. I like it.

**Comment by ffaf1** [comment]
I have made a one-char modification to `init-interactive.test.h`. This way we have a regression test and breakage on future `cabal init` changes is minimised.

**Comment by Mikolaj** [comment]
Wow, that's impressive. I already prepared a budget extension proposal slides to cover the 1000-times larger test and you did it with one character. :D

**Pull request merged**